### PR TITLE
Verify device id before running the simulator

### PR DESCRIPTION
**Pull request by Fatme**

### CODE REVIEW

File: lib/iphone-simulator-xcode-simctl.ts [hunk 1]
```diff
@@ -234,15 +234,15 @@ export class XCodeSimctlSimulator extends IPhoneSimulatorNameGetter implements I
 	}
 
 	public startSimulator(device?: IDevice): void {
-		if (!device || !device.runtimeVersion || !device.fullId) {
-			device = this.getDeviceToRun(device);
-		}
-
 		// In case the id is undefined, skip verification - we'll start default simulator.
 		if (device.id) {
```
Comment (KristianDD) [inline lib/iphone-simulator-xcode-simctl.ts]: Check that device is not undefined `if(device && device.id)` as it is an optional parameter.

File: lib/iphone-simulator-xcode-simctl.ts [hunk 2]
```diff
@@ -234,15 +234,15 @@ export class XCodeSimctlSimulator extends IPhoneSimulatorNameGetter implements I
...
 			this.verifyDevice(device);
 		}
 
+		if (!device || !device.runtimeVersion || !device.fullId) {
+			device = this.getDeviceToRun(device);
```
Comment (KristianDD) [inline lib/iphone-simulator-xcode-simctl.ts]: What happens if we don't have any devices to run? Should we have another check here?

Comment (Fatme) [reply lib/iphone-simulator-xcode-simctl.ts]: Actually `getDeviceToRun` always returns a device so it is not possible device to be null here.

Comment (KristianDD) [reply lib/iphone-simulator-xcode-simctl.ts]: If `this.simctl.getDevices()` returns empty array it might be undefined, but I agree it is unlikely.

**Pull request merged**